src/sentry/replays/consumers/recording/process_recording.py

```diff
@@ -119,19 +120,25 @@ def _store(
                 BytesIO(recording_segment),
                 blob_size=settings.SENTRY_ATTACHMENT_BLOB_SIZE,
             )
-            # associate this file with an indexable replay_id via ReplayRecordingSegment
-            ReplayRecordingSegment.objects.create(
-                replay_id=message_dict["replay_id"],
-                project_id=message_dict["project_id"],
-                segment_id=headers["segment_id"],
-                file_id=file.id,
-            )
+
+            try:
+                # associate this file with an indexable replay_id via ReplayRecordingSegment
+                ReplayRecordingSegment.objects.create(
+                    replay_id=message_dict["replay_id"],
+                    project_id=message_dict["project_id"],
+                    segment_id=headers["segment_id"],
+                    file_id=file.id,
+                )
+            except IntegrityError:
+                # Same message was encountered more than once.
+                logger.warning("Recording-segment has already been processed.")
```

Comment:
could we add the replay_id / project_id / segment_id as extra fields here in the logger?